pkg/solana/txm/txm.go

```diff
@@ -378,135 +453,203 @@ func (txm *Txm) confirm() {
+			txm.lggr.Errorw("no signatures found for expired transaction", "id", tx.UUID)
+			continue
+		}
+		_, err := txm.txs.Remove(tx.signatures[0]) // only picking signature[0] because remove func removes all remaining signatures.
+		if err != nil {
+			txm.lggr.Errorw("failed to remove expired transaction", "id", tx.UUID, "error", err)
+			continue
+		}
+
+		rebroadcastTx := &PendingTx{
+			Tx:               tx.Tx,
+			BalanceCheck:     tx.BalanceCheck,
+			Amount:           tx.Amount,
+			From:             tx.From,
+			IDSetByCaller:    tx.IDSetByCaller,
+			rebroadcastCount: tx.rebroadcastCount + 1,
+		}
+
+		// If the ID was set by the caller, we should keep it similar.
+		if tx.IDSetByCaller {
+			// Append #tx.rebroadcast count to the end of the ID.
+			rebroadcastTx.UUID = fmt.Sprintf("%s#%d", tx.UUID, rebroadcastTx.rebroadcastCount)
+		} else {
+			rebroadcastTx.UUID = uuid.New().String()
+		}
+
+		// Re-enqueue the transaction for rebroadcasting
+		err = txm.Enqueue(ctx, rebroadcastTx)
+		if err != nil {
+			txm.lggr.Errorw("failed to enqueue rebroadcast transaction", "previous id", tx.UUID, "new id", rebroadcastTx.UUID, "error", err)
+			continue
+		}
+
+		txm.lggr.Infow("rebroadcast transaction sent", "previous id", tx.UUID, "new id", rebroadcastTx.UUID)
+	}
+}
+
+func (txm *Txm) processSignatureStatuses(sigs []solanaGo.Signature, res []*rpc.SignatureStatusesResult) {
+	// Sort signatures and results process successful first
+	sortedSigs, sortedRes, err := SortSignaturesAndResults(sigs, res)
+	if err != nil {
+		txm.lggr.Errorw("sorting error", "error", err)
+		return
+	}
+
+	for i := 0; i < len(sortedRes); i++ {
+		sig, status := sortedSigs[i], sortedRes[i]
+		// if status is nil (sig not found), continue polling
+		// sig not found could mean invalid tx or not picked up yet
+		if status == nil {
+			txm.handleNotFoundSignatureStatus(sig)
+			continue
+		}
+
+		// if signature has an error, end polling
+		if status.Err != nil {
+			txm.handleErrorSignatureStatus(sig, status)
+			continue
+		}
+
+		switch status.ConfirmationStatus {
+		case rpc.ConfirmationStatusProcessed:
+			// if signature is processed, keep polling for confirmed or finalized status
+			txm.handleProcessedSignatureStatus(sig)
+			continue
+		case rpc.ConfirmationStatusConfirmed:
+			// if signature is confirmed, keep polling for finalized status
+			txm.handleConfirmedSignatureStatus(sig)
+			continue
+		case rpc.ConfirmationStatusFinalized:
+			// if signature is finalized, end polling
+			txm.handleFinalizedSignatureStatus(sig)
+			continue
+		default:
+			txm.lggr.Warnw("unknown confirmation status", "signature", sig, "status", status.ConfirmationStatus)
+			continue
+		}
+	}
+}
+
+func (txm *Txm) handleNotFoundSignatureStatus(sig solanaGo.Signature) {
+	txm.lggr.Debugw("tx state: not found", "signature", sig)
+
+	// check confirm timeout exceeded
+	if txm.txs.Expired(sig, txm.cfg.TxConfirmTimeout()) {
```

Comment:
Just noticed comparing this logic against the processed method, we should also add the check for TxConfirmTimeout=0 here too
```if txm.cfg.TxConfirmTimeout() != 0*time.Second && txm.txs.Expired(sig, txm.cfg.TxConfirmTimeout()) {```